Update documentation (#221)
OpenLiberty/liberty-tools-intellij: README.md

**Note: This is an early release.**

![Liberty Tools Extension](docs/images/liberty-tool-window-view.png)

This plugin allows you to run your Liberty Maven or Liberty Gradle projects through the Liberty tool window or the IntelliJ "Go to Action" menu. You can start, stop, or interact with [Liberty dev mode](https://openliberty.io/docs/latest/development-mode.html) on all configured [Liberty Maven](https://github.com/OpenLiberty/ci.maven/blob/master/docs/dev.md#dev) and [Liberty Gradle](https://github.com/OpenLiberty/ci.gradle/blob/master/docs/libertyDev.md) projects in your workspace. Liberty Tools help you quickly and easily edit your application and configuration files by providing language support features for MicroProfile, Jakarta EE, and Liberty configuration and Java files.

## Features
- View Liberty projects in the Liberty tool window.
- Start/Stop dev mode.
- Start dev mode with custom parameters.
- Start dev mode in a container.
- Start dev mode with the debugger attached.
- Run tests.
- View test reports.

> Liberty Tools for IntelliJ IDEA consumes the [Liberty Config Language Server](https://github.com/OpenLiberty/liberty-language-server) to provide editing assistance for Liberty server configuration files:

Reviewer: ```suggestion
Liberty Tools for IntelliJ IDEA consumes the [Liberty Config Language Server](https://github.com/OpenLiberty/liberty-language-server) to provide editing assistance for Liberty configuration files:
```

- server.xml
- server.env
- bootstrap.properties

For more information, see the [project documentation in GitHub](https://github.com/OpenLiberty/liberty-language-server#liberty-config-language-server).

![Liberty Config Language Server completion](docs/images/LCLS-server-xml-completion.png)

Liberty Tools for IntelliJ IDEA consumes [Eclipse LSP4Jakarta](https://github.com/eclipse/lsp4jakarta), the Language Server for Jakarta EE, to provide editing assistance for Jakarta EE APIs in Java files. For more information, see the [project documentation in GitHub](https://github.com/eclipse/lsp4jakarta#eclipse-lsp4jakarta).

> Liberty Tools for IntelliJ IDEA consumes [Eclipse LSP4MP](https://github.com/eclipse/lsp4mp), the Language Server for MicroProfile, to provide editing assistance for Jakarta EE APIs in Java and microprofile-config.properties files. For more information, see the [project documentation in GitHub](https://github.com/eclipse/lsp4mp#eclipse-lsp4mp---language-server-for-microprofile).

Reviewer: ```suggestion
Liberty Tools for IntelliJ IDEA consumes [Eclipse LSP4MP](https://github.com/eclipse/lsp4mp), the Language Server for MicroProfile, to provide editing assistance for MicroProfile APIs in Java and microprofile-config.properties files. For more information, see the [project documentation in GitHub](https://github.com/eclipse/lsp4mp#eclipse-lsp4mp---language-server-for-microprofile).
```

| Start                                | Starts dev mode.                                                                                                                                                                                                                                                                                                                                                                                                          |
| Start…                               | Opens the Run Configurations dialog to customize and start dev mode. Supported parameters can be found in the documentation for the [dev goal of the Liberty Maven Plugin](https://github.com/OpenLiberty/ci.maven/blob/master/docs/dev.md#additional-parameters) and the [libertyDev task of the Liberty Gradle Plugin](https://github.com/OpenLiberty/ci.gradle/blob/master/docs/libertyDev.md#command-line-parameters). |

> | Start in a container                 | Starts dev mode with the server in a container. The `liberty-maven-plugin` must be version `3.3-M1` or higher. The `liberty-gradle-plugin` must be version `3.1-M1` or higher.                                                                                                                                                                                                                                            

Reviewer: ```suggestion
| Start in a container                 | Starts dev mode with Liberty running in a container. The `liberty-maven-plugin` must be version `3.3-M1` or higher. The `liberty-gradle-plugin` must be version `3.1-M1` or higher.                                                                                                                                                                                                                                            
```

This plugin allows you to run your Liberty Maven or Liberty Gradle projects through the Liberty tool window or the IntelliJ "Go to Action" menu. You can start, stop, or interact with [Liberty dev mode](https://openliberty.io/docs/latest/development-mode.html) on all configured [Liberty Maven](https://github.com/OpenLiberty/ci.maven/blob/master/docs/dev.md#dev) and [Liberty Gradle](https://github.com/OpenLiberty/ci.gradle/blob/master/docs/libertyDev.md) projects in your workspace. Liberty Tools help you quickly and easily edit your application and configuration files by providing language support features for MicroProfile, Jakarta EE, and Liberty configuration and Java files.

## Features
- View Liberty projects in the Liberty tool window.
- Start/Stop dev mode.
- Start dev mode with custom parameters.
- Start dev mode in a container.
- Start dev mode with the debugger attached.
- Run tests.
- View test reports.
- Editing assistance for Liberty configuration files: server.xml, server.env, bootstrap.properties.
- Editing assistance for Jakarta EE APIs in Java files.
- Editing assistance for MicroProfile APIs in microprofile-config.properties and Java files.

> Feature completion when editing server.xml files:

Reviewer 2: ```suggestion
Feature completion in server.xml files helps you quickly edit your Liberty runtime configuration.

```

![Liberty Config Language Server completion](docs/images/LCLS-server-xml-completion.png)

> Jakarta RESTful Web Services snippets when editing Java files:

Reviewer 2: ```suggestion
Helpful code snippets are provided for Jakarta EE APIs in Java files, such as Jakarta RESTful Web Services :

```

![LSP4Jakarta completion](docs/images/LSP4Jakarta-rest-completion.png)

> Completion when editing microprofile-config.properties files:

Reviewer 2: ```suggestion
Code completion when editing microprofile-config.properties files helps you easily set properties for MicroProfile APIs.

```

![LSP4MP completion](docs/images/LSP4MP-mp-properties-completion.png)

> 

Author: Currently IntelliJ is not supporting diagnostics, only code completion (including snippets).

Editing assistance for Jakarta EE APIs in Java files is provided through [Eclipse LSP4Jakarta](https://github.com/eclipse/lsp4jakarta), the Language Server for Jakarta EE. For more information, see the [project documentation in GitHub](https://github.com/eclipse/lsp4jakarta#eclipse-lsp4jakarta).

Editing assistance for MicroProfile APIs in microprofile-config.properties and Java files is provided through [Eclipse LSP4MP](https://github.com/eclipse/lsp4mp), the Language Server for MicroProfile. For more information, see the [project documentation in GitHub](https://github.com/eclipse/lsp4mp#eclipse-lsp4mp---language-server-for-microprofile).

## Quick Start

- Install [_Liberty Tools_ from the JetBrains Marketplace](https://plugins.jetbrains.com/plugin/14856-liberty-tools).
- Projects with the Liberty Maven Plugin or Liberty Gradle Plugin configured will appear in the Liberty tool window on
  the sidebar. If not enabled by default, the tool window can be viewed by selecting **View > Tool Windows > Liberty**.
- Select a project in the Liberty tool window to view the available commands.

> ## Action Commands

Reviewer 2: is "Action Commands" vendor terminology? If not maybe just "Commands". 

Also, add an introduction for the table:

The following commands are available when you select a project in the Liberty tool window.

Author: "Actions" is more IntelliJ specific terminology. Will go with just "Actions"

| View unit test report (Maven)        | Views the unit test report file if it exists at `/target/site/surefire-report.html`.                                                                                                                                                                                                                                                                                                                                      |
| View test report (Gradle)            | Opens the test report file if it exists at the default location `build/reports/tests/test/index.html`. This action command is only available to Gradle projects. Gradle projects only have a single action command for test result reporting.                                                                                                                                                                             |

## Requirements

> **IntelliJ IDEA**: Starting with the Liberty Tools for IntelliJ IDEA 0.0.8 release, IntelliJ IDEA version 2022.2 is required. Liberty Tools for IntelliJ IDEA is compatible with the Community Edition of IntelliJ IDEA.

Reviewer 2: ```suggestion
Starting with the Liberty Tools for IntelliJ IDEA 0.0.8 release, IntelliJ IDEA version 2022.2 is required. Liberty Tools for IntelliJ IDEA is compatible with the Community Edition of IntelliJ IDEA.
```

**Note: This is an early release.**

![Liberty Tools Extension](docs/images/liberty-tool-window-view.png)

This plugin allows you to run your Liberty Maven or Liberty Gradle projects through the Liberty tool window or the IntelliJ "Go to Action" menu. You can start, stop, or interact with [Liberty dev mode](https://openliberty.io/docs/latest/development-mode.html) on all configured [Liberty Maven](https://github.com/OpenLiberty/ci.maven/blob/master/docs/dev.md#dev) and [Liberty Gradle](https://github.com/OpenLiberty/ci.gradle/blob/master/docs/libertyDev.md) projects in your workspace. Liberty Tools help you quickly and easily edit your application and configuration files by providing language support features for MicroProfile, Jakarta EE, and Liberty configuration and Java files.

## Features
- View Liberty projects in the Liberty tool window.
- Start/Stop dev mode.
- Start dev mode with custom parameters.
- Start dev mode in a container.
- Start dev mode with the debugger attached.
- Run tests.
- View test reports.

> - Editing assistance for Liberty configuration files: server.xml, server.env, bootstrap.properties.

Reviewer 2: ```suggestion
- Code with language assistance in the following configuration and application files:
    - server.xml, server.env, bootstrap.properties Liberty configuration files
    - Jakarta EE 9.x APIs in Java files
    - MicroProfile APIs in microprofile-config.properties and Java files
```

- Editing assistance for Jakarta EE APIs in Java files.
- Editing assistance for MicroProfile APIs in microprofile-config.properties and Java files.

> 

Reviewer 2: ```suggestion

Language assistance features include code completion, hover descriptions, and configuration diagnostics.
```

Reviewer 2: it has the snippets for Jakarta- is that really just a kind of completion? If not, could call out completion and snippets and leave out hover and diagnostics

> <figure>

Reviewer 2: Instead of the caption, I suggest using a standalone sentence with a blank line after preceding the screenshot. GitHub MD is not good at displaying figures and captions in a readable manner

  <img
  src="docs/images/LCLS-server-xml-completion.png"
  alt="Liberty Config Language Server completion">

>   <figcaption align = "center">Feature completion when editing server.xml files</figcaption>

Reviewer 2: sentence suggestion:

Code completion in the server.xml files helps you quickly edit your Liberty runtime configuration.

</figure>

<figure>
  <img
  src="docs/images/LSP4Jakarta-rest-completion.png"
  alt="Eclipse LSP4Jakarta completion">

>   <figcaption align = "center">Jakarta RESTful Web Services snippets when editing Java files</figcaption>

Reviewer 2: sentence suggestion

Helpful pre-configured code snippets are provided for Jakarta EE APIs such as Jakarta RESTful Web Services.

## Features
- View Liberty projects in the Liberty tool window.
- Start/Stop dev mode.
- Start dev mode with custom parameters.
- Start dev mode in a container.
- Start dev mode with the debugger attached.
- Run tests.
- View test reports.
- Code with language assistance in the following configuration and application files:
    - server.xml, server.env, bootstrap.properties Liberty configuration files
    - Jakarta EE 9.x APIs in Java files
    - MicroProfile APIs in microprofile-config.properties and Java files

Feature completion in server.xml files helps you quickly edit your Liberty runtime configuration.

![Liberty Config Language Server completion](docs/images/LCLS-server-xml-completion.png)

Helpful code snippets are provided for Jakarta EE APIs in Java files, such as Jakarta RESTful Web Services :

![LSP4Jakarta completion](docs/images/LSP4Jakarta-rest-completion.png)

Code completion when editing microprofile-config.properties files helps you easily set properties for MicroProfile APIs.

![LSP4MP completion](docs/images/LSP4MP-mp-properties-completion.png)

> Editing assistance for Liberty configuration files is provided through the [Liberty Config Language Server](https://github.com/OpenLiberty/liberty-language-server). For more information, see the [project documentation in GitHub](https://github.com/OpenLiberty/liberty-language-server#liberty-config-language-server).

Reviewer 2: wonder if we can reduce some redundancy here.

Editing assistance for configuration and application files is provided through the following language server projects, which this project consumes. For more information, see the documentation for these projects.

- Liberty configuration files: [Liberty Config Language Server](https://github.com/OpenLiberty/liberty-language-server)
- Jakarta EE APIs in Java files: [Eclipse LSP4Jakarta](https://github.com/eclipse/lsp4jakarta), the Language Server for Jakarta EE
- MicroProfile APIs in microprofile-config.properties and Java files: [Eclipse LSP4MP](https://github.com/eclipse/lsp4mp), the Language Server for MicroProfile

Author: Updated

Editing assistance for Jakarta EE APIs in Java files is provided through [Eclipse LSP4Jakarta](https://github.com/eclipse/lsp4jakarta), the Language Server for Jakarta EE. For more information, see the [project documentation in GitHub](https://github.com/eclipse/lsp4jakarta#eclipse-lsp4jakarta).

Editing assistance for MicroProfile APIs in microprofile-config.properties and Java files is provided through [Eclipse LSP4MP](https://github.com/eclipse/lsp4mp), the Language Server for MicroProfile. For more information, see the [project documentation in GitHub](https://github.com/eclipse/lsp4mp#eclipse-lsp4mp---language-server-for-microprofile).

## Quick Start

- Install [_Liberty Tools_ from the JetBrains Marketplace](https://plugins.jetbrains.com/plugin/14856-liberty-tools).
- Projects with the Liberty Maven Plugin or Liberty Gradle Plugin configured will appear in the Liberty tool window on
  the sidebar. If not enabled by default, the tool window can be viewed by selecting **View > Tool Windows > Liberty**.
- Select a project in the Liberty tool window to view the available commands.

> For detailed instructions on how to use the Liberty actions check out the [Liberty Tools for IntelliJ IDEA User Guide](docs/user-guide.md) page.

Reviewer 2: will the guide include minimum requirements? if so, could call that out here for those that might be looking
```suggestion
For minimum requirements information  and detailed instructions on how to use the Liberty actions, check out the [Liberty Tools for IntelliJ IDEA user guide](docs/user-guide.md) page.
```

# Liberty Tools for IntelliJ IDEA

![JetBrains plugins](https://img.shields.io/jetbrains/plugin/v/14856-liberty-tools.svg?style=for-the-badge)
![Downloads](https://img.shields.io/jetbrains/plugin/d/14856-liberty-tools?style=for-the-badge&)
[![License](https://img.shields.io/badge/License-EPL%202.0-red.svg?style=for-the-badge&label=license&logo=eclipse)](https://www.eclipse.org/legal/epl-2.0/)

Liberty Tools for IntelliJ IDEA is an IntelliJ IDEA plugin for developing cloud-native Java applications with [Open Liberty](https://openliberty.io/) and [WebSphere Liberty](https://www.ibm.com/products/websphere-liberty). Iterate fast with Liberty dev mode, code with assistance for MicroProfile and Jakarta EE APIs, and easily edit Liberty configuration files.

**Note: This is an early release.**

![Liberty Tools Extension](docs/images/liberty-tool-window-view.png)

---
- [Features](#features)

> - [Quick Start](#quick-start)

Reviewer 2: ```suggestion
- [Quick start](#quick-start)
```

- [Actions](#actions)

> - [External Settings](#external-settings)

Reviewer 2: ```suggestion
- [External settings](#external-settings)
```

Helpful code snippets are provided for Jakarta EE APIs in Java files, such as Jakarta RESTful Web Services.

![Eclipse LSP4Jakarta completion](docs/images/LSP4Jakarta-rest-completion.png)

Code completion for `microprofile-config.properties` files helps you easily set properties for MicroProfile APIs.

![Eclipse LSP4MP completion](docs/images/LSP4MP-mp-properties-completion.png)

Editing assistance for configuration and application files is provided through the following language server projects, which this project consumes. For more information, see the documentation for these projects.

- Liberty configuration files: [Liberty Config Language Server](https://github.com/OpenLiberty/liberty-language-server#liberty-config-language-server)
- Jakarta EE APIs in Java files:  [Eclipse LSP4Jakarta](https://github.com/eclipse/lsp4jakarta#eclipse-lsp4jakarta), the Language Server for Jakarta EE.
- MicroProfile APIs in microprofile-config.properties and Java files: [Eclipse LSP4MP](https://github.com/eclipse/lsp4mp#eclipse-lsp4mp---language-server-for-microprofile), the Language Server for MicroProfile.

> ## Quick Start

Reviewer 2: ```suggestion
## Quick start
```

- Install [_Liberty Tools_ from the JetBrains Marketplace](https://plugins.jetbrains.com/plugin/14856-liberty-tools).

> - Projects with the Liberty Maven Plugin or Liberty Gradle Plugin configured will appear in the Liberty tool window on the sidebar. If the tool window is not enabled by default, you can view by selecting **View > Tool Windows > Liberty**.

Reviewer 2: ```suggestion
- - Select a project in the Liberty tool window to view the available commands.

Projects with the Liberty Maven Plugin or Liberty Gradle Plugin configured will appear in the Liberty tool window on the sidebar. If the tool window is not enabled by default, you can view by selecting **View > Tool Windows > Liberty**.
```

| View unit test report (Maven)        | Views the unit test report file if it exists at `/target/site/surefire-report.html`.                                                                                                                                                                                                                                                                                                                                      |
| View test report (Gradle)            | Opens the test report file, if it exists at the `build/reports/tests/test/index.html` default location. This action command is available only to Gradle projects. Gradle projects have only a single action command for test result reporting.                                                                                                                                                                             |

> ## External Settings

Reviewer 2: ```suggestion
## External settings
```

Helpful code snippets are provided for Jakarta EE APIs in Java files, such as Jakarta RESTful Web Services.

![Eclipse LSP4Jakarta completion](docs/images/LSP4Jakarta-rest-completion.png)

Code completion when editing microprofile-config.properties files helps you easily set properties for MicroProfile APIs.

![Eclipse LSP4MP completion](docs/images/LSP4MP-mp-properties-completion.png)

Editing assistance for configuration and application files is provided through the following language server projects, which this project consumes. For more information, see the documentation for these projects.

- Liberty configuration files: [Liberty Config Language Server](https://github.com/OpenLiberty/liberty-language-server#liberty-config-language-server)
- Jakarta EE APIs in Java files:  [Eclipse LSP4Jakarta](https://github.com/eclipse/lsp4jakarta#eclipse-lsp4jakarta), the Language Server for Jakarta EE.
- MicroProfile APIs in microprofile-config.properties and Java files: [Eclipse LSP4MP](https://github.com/eclipse/lsp4mp#eclipse-lsp4mp---language-server-for-microprofile), the Language Server for MicroProfile.

## Quick Start

> - Install [_Liberty Tools_ from the JetBrains Marketplace](https://plugins.jetbrains.com/plugin/14856-liberty-tools).

Reviewer 2: my suggestion was lost I think-
